ts/components/Select.svelte

```diff
@@ -3,18 +3,48 @@ Copyright: Ankitects Pty Ltd and contributors
 License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
 -->
 <script lang="ts">
+    import { altPressed, isArrowDown, isArrowUp } from "@tslib/keys";
     import { createEventDispatcher, setContext } from "svelte";
     import { writable } from "svelte/store";
 
     import { selectKey } from "./context-keys";
     import IconConstrain from "./IconConstrain.svelte";
     import { chevronDown } from "./icons";
     import Popover from "./Popover.svelte";
+    import SelectOption from "./SelectOption.svelte";
     import WithFloating from "./WithFloating.svelte";
 
     // eslint-disable
     type T = $$Generic;
 
+    // E may need to derive content, but we default to them being the same for convenience of usage
+    type E = $$Generic;
+    type C = $$Generic;
+    export let list: E[];
+    export let parser: (item: E) => { content: C; value?: T; disabled?: boolean } = (
+        item,
+    ) => {
+        return {
+            content: item as unknown as C,
+        };
+    };
+    const parsed = list.map(parser).map(({ content, value, disabled = false }, i) => ({
+        content,
+        value: value === undefined ? (i as T) : value,
+        disabled,
+    }));
+    const buttons: HTMLButtonElement[] = Array(list.length);
+    let selected: number | null = null;
+    const last = list.length - 1;
+    const ids = {
+        focused: "focused",
+        popover: "popover",
+    };
+    const roles = {
+        option: "option",
+        popover: "listbox",
+    };
```

Comment:
Please inline the values from `ids` and `roles` in the usage sites, and remove these lines, as using constants for DOM properties is not a pattern that the rest of the codebase uses (and you've been inconsistent with usage here too, using role="combobox" below.